editor/src/components/editor/actions/action-utils.ts

```diff
@@ -213,3 +218,31 @@ export function isFromVSCode(action: EditorAction): boolean {
       return false
   }
 }
+
+export function isClearInteractionSession(action: EditorAction): action is ClearInteractionSession {
```

Comment:
This function should absolutely not be a type guard

```suggestion
export function isClearInteractionSession(action: EditorAction): boolean {
```

Reply:
why not? it's called `isClearInteractionSession` and returns if the action is CLEAR_INTERACTION_SESSION

Reply:
Because it isn't a type guard! Otherwise we're telling TS that a `TRANSIENT_ACTIONS` action or `ATOMIC` action is a `ClearInteractionSession`, when clearly neither of those are.

Reply:
ah yeah you are right about that :D

Reply:
fixed both in 68f29127